units/recalculate_change_points.go

```diff
@@ -105,24 +105,48 @@ func (j *recalculateChangePointsJob) Run(ctx context.Context) {
 		}
 
 		result, err := j.changePointDetector.DetectChanges(ctx, floatSeries)
+		if err != nil {
+			j.AddError(errors.Wrapf(err, "Unable to detect change points in time series %s", j.PerformanceResultId))
+			return
+		}
 
 		var changePoints []model.ChangePoint
-		for _, pointIndex := range result {
-			mapped := model.CreateChangePoint(pointIndex, series.Measurement, j.changePointDetector.Algorithm().Name(), j.changePointDetector.Algorithm().Version(), algorithmConfigurationToOptions(j.changePointDetector.Algorithm().Configuration()))
+		for i, pointIndex := range result {
+			startFirstPartition := 0
+			if i != 0 {
+				startFirstPartition = result[i-1]
+			}
+			endSecondPartition := len(floatSeries)
+			if i != len(result)-1 {
+				endSecondPartition = result[i+1]
+			}
+			percentChange := calculatePercentChange(startFirstPartition, pointIndex, endSecondPartition, floatSeries)
```

Comment:
I feel like this would be clearer:
```
for i, changePoint := range result {
	var lowerBound int
	var upperBound int

	if i == 0 {
		lowerBound = 0
	} else {
		lowerBound = result[i-1]
	}

	if i == len(result)-1 {
		upperBound = len(floatSeries)
	} else {
		upperBound = result[i+1]
	}

	windowBeforeAverage := getAverageValue(floatSeries[lowerBound:changePoint])
	windowAfterAverage := getAverageValue(floatSeries[changePoint:upperBound])

	change := windowAfterAverage / windowBeforeAverage

	percentChange := 100 * (change - 1)
         mapped := ...etc
}
```

What do you think?

Reply:
I've changed it to hopefully be a little more clear and taken most of what you've suggested. What do you think of this?

`			var lowerBound int
			var upperBound int

			if i == 0 {
				lowerBound = 0
			} else {
				lowerBound = result[i-1]
			}

			if i == len(result)-1 {
				upperBound = len(floatSeries)
			} else {
				upperBound = result[i+1]
			}

			lowerWindow := floatSeries[lowerBound:pointIndex]
			upperWindow := floatSeries[pointIndex:upperBound]
			percentChange := calculatePercentChange(lowerWindow, upperWindow)`